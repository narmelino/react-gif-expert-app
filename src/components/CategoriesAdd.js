import { useState } from "react";
import PropTypes from 'prop-types';

const CategoriesAdd = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputValue = (e) => {
    setInputValue(e.target.value);
  }

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories(categories => [inputValue, ...categories]);
      setInputValue('');
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <input
        type='text'
        value={inputValue}
        onChange={handleInputValue}
      />
    </form>
  );
}

CategoriesAdd.propTypes = {
  setCategories: PropTypes.func.isRequired,
}

export default CategoriesAdd;