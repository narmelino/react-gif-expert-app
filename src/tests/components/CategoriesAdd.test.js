import { shallow } from "enzyme";
import CategoriesAdd from '../../components/CategoriesAdd';

describe('Pruebas con CategoriesAdd', () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<CategoriesAdd setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<CategoriesAdd setCategories={setCategories} />)
  });

  test('Debe renderizar correctamente en componente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe cambiar el texto del input', () => {
    const input = wrapper.find('input');
    const value = 'Hola mundo!';
    input.simulate('change', { target: { value } });
  });

  test('No debe postear la info en el submit', () => {
    wrapper.find('form').simulate('submit', { preventDefault(){} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test('Debe postear la info y limpiar el input', () => {
    // Simular que ingresa un valor
    const input = wrapper.find('input');
    const value = 'Hola mundo!';
    input.simulate('change', { target: { value } });

    // Simular que hizo el submit y que la funcion se haya llamado una vez
    wrapper.find('form').simulate('submit', { preventDefault(){} });

    expect(setCategories).toHaveBeenCalled();

    expect(setCategories).toHaveBeenCalledWith(expect.any(Function)); //Determina que setCategories se halla llamado con un argumento de tipo funci√≥n.

    expect(input.prop('value').trim()).toBe('');
  });
});