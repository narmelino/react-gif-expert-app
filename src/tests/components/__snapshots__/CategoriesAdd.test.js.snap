// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con CategoriesAdd Debe renderizar correctamente en componente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
